src/fights.ts

```diff
@@ -1828,10 +1564,22 @@ function deliverThesis(): void {
     outfit("checkpoint");
   }
   cliExecute("gain 1800 muscle");
+
+  let thesisLocation = $location`Uncle Gator's Country Fun-Time Liquid Waste Sluice`;
+  if (thesisInNEP) {
+    thesisLocation = $location`The Neverending Party`;
+  }
+  // if running nobarf, might not have access to Uncle Gator's. Space is cheaper.
+  else if (!(get("stenchAirportAlways") || get("_stenchAirportToday"))) {
```

Comment:
`thesisInNEP` keys off of having NEP fights left--I think it's worth checking the NEP before buying a transponder, because spending a non-free turn in the NEP is still better than spending however much on a transponder (I know they're cheap, but still)